docs/rules/match-component-import-name.md

```diff
@@ -0,0 +1,109 @@
+---
+pageClass: rule-details
+sidebarDepth: 0
+title: vue/match-component-import-name
+description: require the registered component name to match the imported component name
+---
+
+# vue/match-component-import-name
+
+> require the registered component name to match the imported component name
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
+
+This rule reports if the name of a registered component does not match its imported name.
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
+
+## :book: Rule Details
+
+By default, this rule will validate that the imported name is the same casing.
+
+Case can be one of: `"kebab-case"` or `"PascalCase"`
+
+An optional prefix can be provided that must be prepended to all imports.
+
+If you are not registering components, this rule will be ignored.
+
+## :wrench: Options
+
+```json
+{
+  "vue/match-component-import-name": [
+    "error",
+    {
+      "prefix": "prefix-",
+      "case": "kebab-case"
```

Comment:
I don't think the rule proposed in #1786 needs to check the casing. I think the casing check should be implemented by another rule. Alternatively, I think you could add an option to the component-definition-name-casing rule.
https://eslint.vuejs.org/rules/component-definition-name-casing.html

I think it's good to make sure that the match-component-import-name rule is defined with the same name as the converted to the kebab-case or PascalCase.



Also, I don't think the `prefix` option is needed. Why did you think we needed `prefix` option?

Reply:
I need the prefix option for the project I'm working on. We prefix every component name with the prefix `sj-` to indicate that its a component that we own, and to avoid name collisions with other libraries. So, for instance, we would have the import:

```
<script>
import Button from '../Button`

export default {
  components: {
    'sj-button': Button,
    'sj-textarea': () => import('../TextArea')
  }
}
</script>
```

Admittedly, I could refactor our product so the imports have the prefix (`import SjButton from '../Button'`), or move all of our files, but this allows me to turn on `match-component-file-name` and `match-component-import-name` without changing any of my templates. If we don't want to maintain this option for `match-component-import-name` going forward, I can remove it.

Thinking about getting options from component-definition-name-casing or match-component-import-name, I'd prefer to make it so this rule can be configured without using/configuring either of them, so I think I'll implement it so that either case is considered matching, and then add a note to the docs that you should use `component-definition-name-casing` to enforce which case.

Reply:
> Admittedly, I could refactor our product so the imports have the prefix (import SjButton from '../Button'), or move all of our files, but this allows me to turn on match-component-file-name and match-component-import-name without changing any of my templates. If we don't want to maintain this option for match-component-import-name going forward, I can remove it.

Hmm. If the prefix makes sense, I think it makes more sense to prefix the variable as well. (In addition, it makes sense that the file names also match.) I think it's a consistent rule. So I don't think the prefix option is needed. I think rules should focus on consistency over flexibility.